### DISCUSSION TITLE
CTS repair_timing with TNS_END_PERCENT=0 still takes a long time

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#4805] [General]
Shouldn't repair_timing terminate more or less immmediately with TNS_END_PERCENT=0 since only the worst path is repaired?

It takes several minutes in my case.

```
[deleted]
Repair setup and hold violations...
TNS end percent 0
[INFO RSZ-0094] Found 1886 endpoints with setup violations.
```

I'm putting a design in CI before we've had time to rework some pathologically deep levels of logic, so there could be some degenerate case that is slow here.

Without TNS_END_PERCENT, repair takes 3 hours or so... Definitely an improvement with TNS_END_PERCENT, CTS w/repair takes only 10 minutes, say 7 of that in repair_timing.

**Comment by maliberty** [comment]
You still have to find the worst endpoint so my guess is that most of the time is just to computing the timing in order to find the endpoint to repair.
